# Chromosome level *Colaptes auratus* genome assembly 

![Image of Colaptes auratus](https://github.com/jphruska/Colaptes_genome/blob/master/JDM14726.jpg)
Image of *Colaptes auratus auratus*. Photograph taken by Joseph Manthey in Lubbock, Texas. 

This repository documents the steps  Northern Flicker Chromosome Level Assembly, including: (1) characterization of the spatial distribution of coding regions and transposable elements and (2) estimation of Northern Flicker mutation rate from four fold degenerate sites. 

For inquiries regarding any of the analyses implemented in the manuscript, please don't hesitate to contact me at jackphruska@gmail.com

## Data

PacBio, Chromium 10x and Hi-C reads were used to generate the assembly, which totaled 1.378 Gb in size, with a scaffold N50 of 43.948 Mb
and a scaffold L50 of 11. The raw reads are available at NCBI (BioProject PRJNA616131). 

## Assembly, Scaffolding and Polishing 

Details of the assembly, polishing and scaffolding of the *C.auratus* genome are detailed in the **Assembly** folder of this repository. In brief, we assembled PacBio contigs with Canu, followed by polishing with chromium 10x reads (fixing ambiguous bases and indels, gap filling, etc) with Pilon. We scaffolded the improved PacBio assembly (increasing scaffold sizes) with chromium 10x reads and Arcs. Lastly, further scaffolding of the assembly was carried out with Hi-C reads and the 3D-DNA pipeline.  

## Synteny 

To identify the chromosomes, we took advantage of the strong synteny in birds and mapped the chromosomes to a Chicken (*Gallus gallus*) genome assembly (NCBI GCF_000002315.6) using MUMMER. Using alignment information generated by MUMMER as input, we created a synteny plot (Figure 1B of manuscript). The relative order of mapping was used to rename the chromosomes. Only the chromosomes that showed strong synteny with *Gallus* assembly were renamed. More details and scripts are provided in the **Synteny_mapping** folder of this repository. 

## Repeats 

We annotated transposable elements and repetitive content using a *de novo* repeat library and RepBase vertebrate database v 25.03. We created a custom repeat library with an early version of the genome assembly (post Arcs scaffolding) and from other Manthey Lab Projects. We generated the library using RepeatModeler, blastn and MAFFT (as implemented in Geneious) to generate consensus sequences for each repetitive element. We summarized repetitive content with RepeatMasker, and plotted the distribution of repeats across 100 kb non-overlapping windows of the genome. More details and scripts are provided in the **Repeats** folder of this repository. 

## CDS

We annnotated CDS regions using MAKER, along with the custom repeat library and protein datasets from Downy Woodpecker *Picoides pubescens* and representatives of three closely related orders (Trogoniformes, Coraciiformes, and Bucerotiformes). After training SNAP and Augustus, we ran MAKER a second time to annotate genes. We plotted the distribution of cds regions across 100 kb non-overlapping windows of the genome. To estimate a *Colaptes* specific mutation rate we extracted four-fold degenerate sites found across alignments of CDS regions across Piciformes, Trogoniformes, Coraciiformes and Bucerotiformes and estimated branch lengths with a user-specified tree (Jarvis et al. 2014). Additional details and scripts are provided in the **CDS** folder of this repository. 





